opds_import.py

```diff
@@ -157,60 +158,79 @@ def import_from_feed(self, feed, even_if_no_author=False,
 
         imported = []
         for metadata in metadata_objs:
-            # Locate or create a LicensePool for this book.
-            license_pool, is_new_license_pool = metadata.license_pool(self._db)
-
-            # Locate or create an Edition for this book.
-            edition, is_new_edition = metadata.edition(self._db)
-
-            if (cutoff_date 
-                and not is_new_license_pool 
-                and not is_new_edition
-                and metadata.circulation 
-                and metadata.circulation.first_appearance < cutoff_date
-            ):
-                # We've already imported this book, we've been told
-                # not to bother with books that appeared before a
-                # certain date, and this book did in fact appear
-                # before that date. There's no reason to do anything.
-                continue
+            try:
+                edition = self.import_from_metadata(
+                    metadata, even_if_no_author, cutoff_date, immediately_presentation_ready
+                )
+                if edition:
+                    imported.append(edition)
+            except Exception, e:
+                # Rather than scratch the whole import, treat this as a failure that only applies
+                # to this item.
+                identifier, ignore = metadata.primary_identifier.load(self._db)
+                message = StatusMessage(500, "Local exception during import:\n%s" % traceback.format_exc())
+                messages[identifier.urn] = message
+
+        return imported, messages, next_links
+
+
+    def import_from_metadata(
```

Comment:
All this 'new' code is just the old inner loop from import_from_metadata(), moved into its own method and dedented.